src/portfft/descriptor_validate.hpp

```diff
@@ -0,0 +1,217 @@
+/***************************************************************************
+ *
+ *  Copyright (C) Codeplay Software Ltd.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's portFFT
+ *
+ **************************************************************************/
+
+#ifndef PORTFFT_DESCRIPTOR_VALIDATE_HPP
+#define PORTFFT_DESCRIPTOR_VALIDATE_HPP
+
+#include <string_view>
+
+#include "common/exceptions.hpp"
+#include "enums.hpp"
+#include "utils.hpp"
+
+namespace portfft::detail::validate {
+
+inline void validate_lengths(const std::vector<std::size_t>& lengths) {
+  if (lengths.empty()) {
+    throw invalid_configuration("Invalid lengths, must have at least 1 dimension");
+  }
+  for (std::size_t i = 0; i < lengths.size(); ++i) {
+    if (lengths[i] == 0) {
+      throw invalid_configuration("Invalid lengths[", i, "]=", lengths[i], ", must be positive");
+    }
+  }
+}
+
+inline void validate_layout(const std::vector<std::size_t>& lengths, detail::layout forward_layout,
```

Comment:
I suppose the layout is also the part of the descriptor, and not the committed descriptor correct ?
In that case let's keep these checks (this and the validate_lengths) in the constructor of descriptor itself. This way we needn't build the kernels before throwing `invalid_configuration`

Reply:
After going through the remaining checks, I suppose quite a few can be moved to the descriptor. 
I am of the opinion that we should not allow one to commit an invalid configuration

Reply:
As above, this is part of the constructor for `committed_descriptor`, so all the code in descriptor_validate.hpp is called before we try to build any kernel bundles.
The committed_descriptor class is big enough in my opinion!

Reply:
Combining this with [this])(https://github.com/codeplaysoftware/portFFT/pull/136#discussion_r1473326926)